doc/release/1.18.0-notes.rst

```diff
@@ -14,6 +14,11 @@ New functions
 Deprecations
 ============
 
+* The `numpy.compat.is_pathlib_like` function is deprecated. For Python 3.6 and
+  above, users are encouraged to use ``isinstance(obj, os.PathLike)``. For code
+  supporting Python 3.5, the snippet ``isinstance(obj,
+  numpy.compat.os_PathLike)`` can be used.
+
```

Comment:
This should now go into a file in `doc/release/upcoming_changes`

Reply:
Thanks.